src/main/scala/carldata/series/TimeSeries.scala

```diff
@@ -185,12 +185,21 @@ case class TimeSeries[V: math.Numeric](idx: Vector[LocalDateTime], ds: Vector[V]
 
   /** Rolling window operation */
   def rollingWindow(windowSize: Duration, f: Seq[V] => V): TimeSeries[V] = {
-    val rs = index.map { t =>
-      val window = slice(t.minus(windowSize), t.plusNanos(1))
-      f(window.values)
+    val rs: mutable.ListBuffer[V] = ListBuffer()
```

Comment:
This function is simpler then previous one.
Also it is also preferred to use immutable data structures.
Could you redesign this function in this way that it won't use ListBuffer?

Reply:
Fixed.